src/Form/StrawberryfieldToolsForm.php

```diff
@@ -0,0 +1,157 @@
+   *   The renderer service.
+   * @param \Drupal\Core\Entity\EntityRepositoryInterface $entity_repository
+   *   The entity repository.
+   */
+  public function __construct(RendererInterface $renderer, EntityRepositoryInterface $entity_repository) {
+    $this->entityRepository = $entity_repository;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('renderer'),
+      $container->get('entity.repository')
+    );
+  }
+
+  public function getFormId() {
+    return 'strawberryfield_tools_form';
+  }
+
+  public function buildForm(array $form, FormStateInterface $form_state, NodeInterface $node = NULL) {
+
+    // For code Mirror
+    // @TODO make this module dependant
+    $settings['mode'] = 'application/ld+json';
+    $settings['readOnly'] = TRUE;
+    $settings['toolbar'] = FALSE;
+    $settings['lineNumbers'] = TRUE;
+
+    if ($sbf_fields = \Drupal::service('strawberryfield.utility')->bearsStrawberryfield($node)) {
+      foreach ($sbf_fields as $field_name) {
+        /* @var $field \Drupal\Core\Field\FieldItemInterface */
+        $field = $node->get($field_name);
+        if (!$field->isEmpty()) {
+          /** @var $field \Drupal\Core\Field\FieldItemList */
+          foreach ($field->getIterator() as $delta => $itemfield) {
+            // Note: we are not longer touching the metadata here.
+            /** @var $itemfield \Drupal\strawberryfield\Plugin\Field\FieldType\StrawberryFieldItem */
+            $json = json_encode(json_decode($itemfield->value), JSON_PRETTY_PRINT);
+            $form_state->set('itemfield', $itemfield);
+            $form['test_jmespath'] = [
+              '#type' => 'textfield',
+              '#default_value' => $form_state->getValue('test_jmespath'),
+              '#title' => $this->t('JMESPATH'),
+              '#description' => $this->t(
+                'Evaluate a JMESPath Query against this ADO\'s JSON. See <a href=":href" target="_blank">JMESPath Tutorial</a>.',
+                [':href' => 'http://jmespath.org/tutorial.html']
+              ),
+
+              '#ajax' => [
+                'callback' => [$this, 'callJmesPathprocess'],
+                'event' => 'change',
+                'keypress' => FALSE,
+                'disable-refocus' => FALSE,
+                'progress' => [
+                  // Graphic shown to indicate ajax. Options: 'throbber' (default), 'bar'.
+                  'type' => 'throbber',
+                ],
+              ],
+              '#required' => TRUE,
+              '#executes_submit_callback' => TRUE,
+              '#submit' =>  ['::submitForm']
+            ];
+            $form['test_jmespath_input'] = [
+              '#type' => 'codemirror',
+              '#codemirror' => $settings,
+              '#default_value' => $json,
+              '#rows' => 15,
+            ];
+            $form['test_output'] = [
```

Comment:
Please move the output to before the actual Input

Reply:
Resolved by 50b5e743edb9cc7639263086697a9586340a4aef